conbench/entities/commit.py

```diff
@@ -287,7 +310,12 @@ def get_commit(self, name, sha):
             response = self.test_commit(sha)
         else:
             url = f"{GITHUB}/repos/{name}/commits/{sha}"
+            # _get_response() may raise an exception, for example if the GH
+            # HTTP API returned a non-2xx HTTP response (e.g. in case of rate
+            # limiting).
             response = self._get_response(url)
+
+        # Note(JP): I think now `response` can never be just `None`.
         return self._parse_commit(response) if response else None
```

Comment:
I think `self._get_response(url)` can't be None now, but `self.test_commit(sha)` can be None due to this line:

https://github.com/conbench/conbench/blob/c029a3d9ca5b3099f87b3983b8f420e46f3d8dd8/conbench/entities/commit.py#L419-L420

Wonder if we should change that line to `raise requests.exceptions.RequestException` to more correctly mock the new behavior? Then this line can just be `return self._parse_commit(response)`.

Reply:
Trying to make sense of this.

The relationship between test_commit() and parse_commit() and get_commit() is mind-bending, in particular without code comments. :)

Reply:
> `raise requests.exceptions.RequestException`

Good idea to raise an exception, but it's IMO for now better to not raise an exception that otherwise only `requests` is supposed to raise. 

I have added a commit with a bit of commentary and also with
```
raise Exception("_mocked_get_response(): simulate _get_response() error")
```
which makes use of the fact that so far we're indeed using overgeneralized exception handlers.